modules/context/api.go

```diff
@@ -140,3 +143,28 @@ func ReferencesGitRepo() macaron.Handler {
 		}
 	}
 }
+
+// 404 handler for APIContext
+// String will replace message, errors will be added to a slice
+func (ctx *APIContext) NotFound(objs ...interface{}) {
+	var message = "Not Found"
+	var errors []string
+	for _, obj := range objs {
+		if err, ok := obj.(error); ok {
+			errors = append(errors, err.Error())
+		} else {
+			message = obj.(string)
+		}
+	}
+
+	swagger, err := url.Parse(setting.AppURL)
+	if err != nil {
+		ctx.Error(500, "Invalid AppURL", err)
+	}
+	swagger.Path = path.Join(swagger.Path, "api", "swagger")
```

Comment:
Why not just use this?
```suggestion
	swagger.Path = path.Join(setting.AppURL, "api", "swagger")
```

Reply:
Without parsing the URL, the `path.Join` call was replacing the double slash in `http(s)://` with a single slash

Reply:
yeah right, now I remember... Only thing may be rename `swagger` variable name to something like `u` or `appURL`